src/events/interactionCreate.test.ts

```diff
@@ -0,0 +1,217 @@
+// Dependencies
+import type { Interaction, CommandInteraction, TextBasedChannel } from 'discord.js';
+import { ChannelType } from 'discord.js';
+
+// Mock allCommands to isolate our test code
+const mockAllCommands = new Map<string, Command>();
+jest.mock('../commands', () => ({
+	allCommands: mockAllCommands,
+}));
+
+// Create two mock commands to track handler behavior
+const mockGlobalExecute = jest.fn();
```

Comment:
To make sure I'm understanding, is interactionCreate an event specifically dealing with commands?

Reply:
Yep. [Here](https://discordjs.guide/creating-your-bot/event-handling.html#event-handling) are some docs on that